### DISCUSSION TITLE
Artifacts Function - Can't get working

### DISCUSSION THREAD

**Discussion by hchasens** [open-webui/open-webui#4220] [Q&A]
Side Note: It'd be great to add a section for documentation or a readme.md to functions and tools.

I haven't been able to get this function working. Any suggestions? 

https://openwebui.com/f/atgehrhardt/artifacts

I've enabled it but I can't get it to render SVG, Javascript, or HTML.

**Comment by justinh-rahb** [comment]
What model are you using it with? We've had best results with Claude 3.5 and Gpt-4o

**Reply by hchasens** [reply]
I've been using Ollama and have tried it with Llama3.1:8b and Phi3-Mini. My understanding is that it should be model-agnostic though. From scanning the code it should just take any markdown code blocks labelled as HTML, javascript, CSS, or SVG then render it. Am I wrong in thinking this?

**Reply by boehm-e** [reply]
to make it work, I had to enable it in `/workspace/functions` and click on the three dots "..." to enable `Global` switch

**Reply by justinh-rahb** [reply]
@boehm-e that's ***a*** way to enable it, but that'll add the capability to all of your models, some of which may not use it effectively. It's better to enable the filter in each model you'd like to use it with in **Workspace > Models**.
